File: array/1629_slowest_key/1629_slowest_key.go
package main

import "fmt"

func slowestKey(releaseTimes []int, keysPressed string) byte {
	t := releaseTimes[0]
	ans := keysPressed[0]
	for i := 1; i < len(releaseTimes); i++ {
		if releaseTimes[i]-releaseTimes[i-1] >= t {
			ans = keysPressed[i]
			t = releaseTimes[i] - releaseTimes[i-1]
		}
	}
	return ans
}

func main() {
	releaseTimes := []int{9, 29, 49, 50}
	keysPressed := "cbcd"
	res := slowestKey(releaseTimes, keysPressed)
	fmt.Println(res)
}

File: array/1629_slowest_key/1629_slowest_key.py
from typing import List

class Solution:
    def slowestKey(self, releaseTimes: List[int], keysPressed: str) -> str:
        n = len(releaseTimes)
        last = longest = releaseTimes[0]
        ans = keysPressed[0]
        for i in range(1, n):
            last_time = releaseTimes[i] - last
            if last_time > longest:
                longest = last_time
                ans = keysPressed[i]
            elif last_time == longest:
                ans = max(ans, keysPressed[i])
            last = releaseTimes[i]
        return ans


releaseTimes = [9,29,49,50]
keysPressed = "cbcd"
res = Solution().slowestKey(releaseTimes, keysPressed)
print(res)